search-commons/src/main/java/no/unit/nva/search2/resource/ResourceQuery.java

```diff
@@ -288,26 +289,69 @@ private MatchQueryBuilder containsPublishedFileQuery() {
         return QueryBuilders.matchQuery(jsonPath(ASSOCIATED_ARTIFACTS, TYPE, KEYWORD), PUBLISHED_FILE);
     }
 
-    private Stream<Entry<ResourceParameter, QueryBuilder>> excludeSubunitsQuery() {
+    public Stream<Entry<ResourceParameter, QueryBuilder>> createSubunitsQuery() {
+        var viewingScope = getViewingScope();
+        if (viewingScopeIsProvided(viewingScope)) {
+            var shouldExcludeSubunits = getExcludeSubunitsKey();
+            return kQueryTools.queryToEntry(VIEWING_SCOPE,createSubunitQuery(shouldExcludeSubunits, viewingScope));
+        } else {
+            return null;
+        }
+    }
+
+    private static QueryBuilder createSubunitQuery(Boolean shouldExcludeSubunits, List<String> viewingScope) {
+        return shouldExcludeSubunits ? excludeSubunitsQuery(viewingScope) : includeSubunitsQuery(viewingScope);
+    }
+
+    private static boolean viewingScopeIsProvided(List<String> viewingScope) {
+        return !viewingScope.isEmpty();
+    }
 
-        var shouldExcludeSubunits = getValue(EXCLUDE_SUBUNITS).asBoolean();
-        var viewingScope = getValue(VIEWING_SCOPE).split(COMMA);
+    private List<String> getViewingScope() {
+        var viewingScopeKeys = getViewingScopeParameters();
+        var viewingScopeParams = viewingScopeKeys.stream().map(searchParameters::get).collect(Collectors.joining(","));
+        return !viewingScopeKeys.isEmpty()
+                   ? extractViewingScope(viewingScopeParams)
+                   : List.of();
+    }
+
+    private List<ResourceParameter> getViewingScopeParameters() {
+        return searchParameters.keySet().stream()
+                   .filter(this::isOrganization)
+                   .toList();
+    }
 
-        var queryBuilder = shouldExcludeSubunits
-            ? excludeSubunitsQuery(viewingScope)
-            : includeSubunitsQuery(viewingScope);
+    public boolean isOrganization(ResourceParameter resourceParameter) {
+        return ResourceParameter.TOP_LEVEL_ORGANIZATION.name().equals(resourceParameter.name())
+               || ResourceParameter.UNIT.name().equals(resourceParameter.name());
+    }
+
+    private static List<String> extractViewingScope(String viewingScopeParameter) {
+        return Arrays.stream(viewingScopeParameter.split(COMMA))
+                   .map(value -> URLDecoder.decode(value, StandardCharsets.UTF_8))
+                   .toList();
+    }
```

Comment:
use key, not string value and use 
getValue(key) and isPresent(key)

Reply:
It seems like entries I am getting here are not decoded:

private List<String> getViewingScope() {
        return Stream.of(getValue(TOP_LEVEL_ORGANIZATION), getValue(UNIT))
                   .filter(asType -> isPresent(asType.getKey()))
                   .map(AsType::as)
                   .map(Objects::toString)
                   .map(value -> URLDecoder.decode(value, StandardCharsets.UTF_8))
                   .toList();
    }
    
    So I have to run 
                       .map(value -> URLDecoder.decode(value, StandardCharsets.UTF_8))

Reply:
Ah, I know what has happened...
by setting them as CUSTOM, decoding doesn't happen...
Now, there are propably no downside of decoding CUSTOM by default.

Reply:
Another thing to note is that we need to handle all items that we set as CUSTOM, and we need to raise an error for items not handled...